test/integration/seaport/seaport-integration.js

```diff
@@ -0,0 +1,205 @@
+
+    // Cut the protocol handler facets into the Diamond
+    await deployAndCutFacets(protocolDiamond.address, facetsToDeploy, maxPriorityFeePerGas);
+
+    const seller = mockSeller(assistant.address, assistant.address, assistant.address, assistant.address);
+
+    const emptyAuthToken = mockAuthToken();
+    const voucherInitValues = mockVoucherInitValues();
+    await accountHandler.connect(assistant).createSeller(seller, emptyAuthToken, voucherInitValues);
+
+    const disputeResolver = mockDisputeResolver(DR.address, DR.address, DR.address, DR.address, true);
+
+    const disputeResolverFees = [new DisputeResolverFee(ethers.constants.AddressZero, "Native", "0")];
+    const sellerAllowList = [];
+
+    await accountHandler.connect(DR).createDisputeResolver(disputeResolver, disputeResolverFees, sellerAllowList);
+
+    const { offer, offerDates, offerDurations, disputeResolverId } = await mockOffer();
+    offer.quantityAvailable = 10;
+
+    await offerHandler
+      .connect(assistant)
+      .createOffer(offer.toStruct(), offerDates.toStruct(), offerDurations.toStruct(), disputeResolverId, "0");
+
+    const voucherAddress = calculateContractAddress(accountHandler.address, seller.id);
+    bosonVoucher = await ethers.getContractAt("BosonVoucher", voucherAddress);
+
+    // Pool needs to cover both seller deposit and price
+    const pool = ethers.BigNumber.from(offer.sellerDeposit).add(offer.price);
+    await fundsHandler.connect(assistant).depositFunds(seller.id, ethers.constants.AddressZero, pool, {
+      value: pool,
+    });
+
+    // Pre mint range
+    await offerHandler.connect(assistant).reserveRange(offer.id, offer.quantityAvailable, bosonVoucher.address);
+    await bosonVoucher.connect(assistant).preMint(offer.id, offer.quantityAvailable);
+
+    // Create seaport offer which tokenId 1
+    const endDate = "0xff00000000000000000000000000";
+    const seaportOffer = seaportFixtures.getTestVoucher(1, bosonVoucher.address, 1, 1);
+    const consideration = seaportFixtures.getTestToken(0, undefined, 1, 2, bosonVoucher.address);
+    ({ order, orderHash, value } = await seaportFixtures.getOrder(
+      bosonVoucher,
+      undefined,
+      [seaportOffer],
+      [consideration],
+      0, // full
+      0,
+      endDate
+    ));
+
+    const orders = [objectToArray(order)];
+    calldata = seaport.interface.encodeFunctionData("validate", [orders]);
+  });
+
+  it("Voucher contract can be used to call seaport validate", async function () {
+    const tx = await bosonVoucher.connect(assistant).callExternalContract(seaport.address, calldata);
+    const receipt = await tx.wait();
+
+    const [, orderParameters] = getEvent(receipt, seaport, "OrderValidated");
+
+    assert.deepEqual(orderParameters, objectToArray(order.parameters));
+  });
+
+  it("Seaport is allowed to transfer vouchers", async function () {
+    await bosonVoucher.connect(assistant).callExternalContract(seaport.address, calldata);
```

Comment:
same calldata is used in previous test step.
Maybe a problem as the token has already been listed.
I would rather create another order for another token to avoid any conflict between the 2 test steps...

Reply:
These tests run independently, even the calldata being the same, when the second test runs, the blockchain doesn't have the state from the first test